plugin/go.vim

```diff
@@ -91,15 +114,16 @@ function! s:GoInstallBinaries(updateBinaries)
 
     if !executable(bin) || a:updateBinaries == 1
       if a:updateBinaries == 1
-        echo "vim-go: Updating ". basename .". Reinstalling ". pkg . " to folder " . go_bin_path
+        echo "vim-go: Updating ". basename .". Reinstalling ". pkg[0] . " to folder " . go_bin_path
       else
-        echo "vim-go: ". basename ." not found. Installing ". pkg . " to folder " . go_bin_path
+        echo "vim-go: ". basename ." not found. Installing ". pkg[0] . " to folder " . go_bin_path
       endif
 
-
-      let out = go#util#System(cmd . shellescape(pkg))
+      let out = go#util#System(cmd . 
+            \ (len(pkg) > 1 ? get(pkg[1], l:platform, '') : '') . ' ' .
+            \ shellescape(pkg[0]))
       if go#util#ShellError() != 0
-        echo "Error installing ". pkg . ": " . out
+        echo "Error installing ". pkg[0] . ": " . out
```

Comment:
I get why `pkg[0]` and `pkg[1]` are used here, but it it's pretty far from s:packages. For readability, it's probably worth naming giving them meaningful names by assigning their values to variables at the start of the `for` loop. Something like:
```
let l:pkgpath = pkg[0]
let l:gogetflags = (len(pkg) > 1 ? get(pkg[1], l:platform, '') : '')
```

And then use those instead of `pkg[0]` and `pkg[1]`.

Reply:
Thanks; updated!